osu.Game.Rulesets.Taiko/Mods/TaikoModClassic.cs

```diff
@@ -3,21 +3,200 @@
+        {
+            public ClassicDrumRoll(DrumRoll original)
+            {
+                StartTime = original.StartTime;
+                Samples = original.Samples;
+                EndTime = original.EndTime;
+                Duration = original.Duration;
+                TickRate = original.TickRate;
+                RequiredGoodHits = original.RequiredGoodHits;
+                RequiredGreatHits = original.RequiredGreatHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollJudgement();
+
+            protected override void CreateTicks(CancellationToken cancellationToken)
+            {
+                if (TickSpacing == 0)
+                    return;
+
+                bool first = true;
+
+                for (double t = StartTime; t < EndTime + TickSpacing / 2; t += TickSpacing)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    AddNested(new ClassicDrumRollTick
+                    {
+                        FirstTick = first,
+                        TickSpacing = TickSpacing,
+                        StartTime = t,
+                        IsStrong = IsStrong
+                    });
+
+                    first = false;
+                }
+            }
+        }
+
+        private class ClassicDrumRollTick : DrumRollTick
+        {
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollTickJudgement();
+        }
+
+        private class ClassicSwell : Swell
+        {
+            public ClassicSwell(Swell original)
+            {
+                StartTime = original.StartTime;
+                Samples = original.Samples;
+                EndTime = original.EndTime;
+                Duration = original.Duration;
+                RequiredHits = original.RequiredHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicSwellJudgement();
+        }
+
+        private class TaikoClassicDrumRollJudgement : TaikoDrumRollJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class TaikoClassicDrumRollTickJudgement : TaikoDrumRollTickJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class TaikoClassicSwellJudgement : TaikoSwellJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class ClassicDrawableDrumRoll : DrawableDrumRoll
+        {
+            public override bool DisplayResult => false;
+
+            protected override void CheckForResult(bool userTriggered, double timeOffset)
+            {
+                if (userTriggered)
+                    return;
+
+                if (timeOffset < 0)
+                    return;
+
+                ApplyResult(r => r.Type = HitResult.IgnoreMiss);
```

Comment:
This is even weirder. So the drum roll's `MaxResult` is `LargeBonus` but it only ever gets `IgnoreMiss`? At this point you could (and probably should) be using `IgnoreHit` in both places.